test/spec/03/getSites.spec.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * getSites handler tests
+ */
+
+import * as sinon from 'sinon'
+import { getSites } from '../../../src/soap/handlers/GetSites.js'
+import { createHelpers } from '../../../src/lib/helpers.js'
+
+// Probably should be moved to a utils?
+const SERVICE_1_1 = 'cuahsi_1_1.asmx'
+
+describe('GetSites handlers', function () {
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('should handle getSites with no station and no parameters', async function () {
+    const method = 'GetSites'
+    const parameters = {
+      site: {
+        string: ['', '']
+      },
+      authToken: ''
+    }
+    const request = { params: { org: '' } }
+    const cache = {}
+    const webAPI = {}
+    const helpers = createHelpers({ cache, webAPI })
+    const ctx = {
+      cache,
+      helpers,
+      service: SERVICE_1_1
+    }
+
+    // For getSites, we need to fake out findMany() and orgId()
+    const findManyFake = sinon.fake.returns([])
+    const orgIdFake = sinon.fake.returns('')
+    sinon.replace(helpers, 'findMany', findManyFake)
+    sinon.replace(helpers, 'orgId', orgIdFake)
+
+    const gen = getSites(request, Object.assign({ method, parameters }, ctx))
+
+    // Followup with with expect statements to check yielded results
+    expect((await gen.next()).value).to.equal(
+      '<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><soap:Body><GetSitesResponse xmlns="http://www.cuahsi.org/his/1.1/ws/"><GetSitesResult>&lt;sitesResponse xmlns=&quot;http://www.cuahsi.org/waterML/1.1/&quot;&gt;&lt;queryInfo&gt;&lt;creationTime&gt;2022-08-31T16:43:39.333Z&lt;/creationTime&gt;&lt;criteria MethodCalled=&quot;GetSites&quot;&gt;&lt;/criteria&gt;&lt;/queryInfo&gt;'
+    )
+    expect((await gen.next()).value).to.equal(
+      '&lt;/sitesResponse&gt;</GetSitesResult></GetSitesResponse></soap:Body></soap:Envelope>'
+    )
+  })
+
+  it('should handle getSites with no station and parameters', async function () {
+    const method = 'GetSites'
+    const parameters = {
+      site: {
+        string: ['string', 'string']
```

Comment:
Maybe use dummy data `string1` and `string2` to ensure the order is preserved in the result.